net/client.go

```diff
@@ -2,6 +2,7 @@ package net
 
 import (
 	"context"
+	"errors"
```

Comment:
These are all changes @sanderpick found to fix multi peer data sync bugs we were seeing.